documentation/src/editor.dox

```diff
@@ -3,23 +3,345 @@
+the "Close" button in the window frame.
+
+Congratulations, you've just built a minimal FLTK app.
+
+
+\section editor_main_menu Chapter 2: Adding a Menu Bar
+
+In this chapter we will add the main menu bar with a File menu
+and a Quit button. This is a good time to define a flag that will
+track changes in the text later.
+
+\code
+void tut2_build_app_menu_bar() {
+  app_window->begin();
+  app_menu_bar = new Fl_Menu_Bar(0, 0, app_window->w(), 25);
+  app_menu_bar->add("File/Quit Editor", FL_COMMAND+'q', menu_quit_callback);
+  app_window->callback(menu_quit_callback);
+  app_window->end();
+}
+\endcode
+
+`begin()` tells FLTK to add all widgets that are created hereafter
+will be added to our app_window. In this particular case, it is redundant
+because creating the window in the previous chapter called `begin()`
+for us.
+
+In the next line, we create the menu bar and add our first menu item
+to it. Menus can be built like file paths with forward slashes '/'
+separating the sub menus from menu items.
+
+Our basic callback is simple:
+\code
+void menu_quit_callback(Fl_Widget *, void *) {
+  Fl::hide_all_windows();
+}
+/endcode
+`Fl::hide_all_windows();` will tell `Fl::run()` to return and `main`
+to exit. I added a dialog box in the app that verifies if the user
+really wants to quit, even if text was changed.
+
+Assigning the window callback allows the `menu_quit_callback` to
+also manage the pressing of the Escape key in the app window. This callback
+is also the perfect time to remind the user if there are unsaved changes in
+our text, so let's add a flag for that and a popup dialog box in the callback.
+
+\code
+bool text_changed = false;
+...
+void menu_quit_callback(Fl_Widget *, void *) {
+  if (text_changed) {
+    int c = fl_choice("Changes in your text have not been saved.\n"
+                      "Do you want to quit the editor anyway?",
+                      "Quit", "Cancel", NULL);
+    if (c == 1) return;
+  }
+  Fl::hide_all_windows();
+}
+\endcode
+
+
+\section editor_text_widget Chapter 3: Adding a Text Editor widget
+
+To keep things simple our text editor will use the
+Fl_Text_Editor widget to edit the text.
+So that we can keep track of changes to the file, we also want to add
+a "modify" callback:
+
+\code
+  app_text_buffer = new Fl_Text_Buffer();
+  app_text_buffer->add_modify_callback(text_changed_callback, NULL);
+  app_editor = new Fl_Text_Editor(0, app_menu_bar->h(),
+    app_window->w(), app_window->h() - app_menu_bar->h());
+  app_editor->buffer(app_text_buffer);
+  app_window->resizable(app_editor);
+\endcode
+
+By setting the resizable, we ensure that resizing the window will
+resize the text editor as well, but restrict the menu bar to resize
+horizontally only.
+
+The text modify callback sets our `text_changed` flag:
+
+\code
+void text_changed_callback(int, int, int, int, const char*, void*) {
+  text_changed = true;
+}
+\endcode
+
+Lastly, I added a "File/New" menu item to the main menu and gave
```

Comment:
Suggested change: "Lastly, we add a "File/New" menu item to the main menu and give"

Note: "we" and present tense (my personal opinion).